### ISSUE TITLE
high cpu usage when opening a connection

### ISSUE THREAD

**Issue by vgoklani** [thefab/tornadis#5]
Hi,

What's the proper way of opening a redis connection? I've noticed that when I run the below, my cpu usage spikes to 100% and remains there. Is there a way of only opening this connection once, as opposed to each time I hit the handler?

```
client = tornadis.Client()
class Handler(tornado.web.RequestHandler):
    @tornado.gen.coroutine
    def get(self):
        data = get_data()
        self.write(data)
        self.finish()

@tornado.gen.coroutine
def get_data()
      yield client.connect()
      data = yield client.call( ... )
      raise tornado.gen.Return( data )
```

**Comment by thefab** [comment]
fixed in master or in 0.2.1 release a few days ago

**Comment by vgoklani** [comment]
I already upgraded to 0.2.1, and the CPU is still sitting at 100%

**Comment by thefab** [comment]
you have to use:

data = yield get_data()

no ?

If you can provide a simple example we can execute here, we would be happy to try to reproduce (and fix) your problem

**Comment by vgoklani** [comment]
Thanks for your help! Please see below for a basic good example. 

Am I opening the connection correctly?

```
import sys
from bson import json_util
import tornado.web
import tornadis
import tornado.httpserver, tornado.wsgi, tornado.ioloop, tornado.options, tornado.web, tornado.websocket
import redis
from tornado.options import define, options

define("port", default=8881, help="run on the given port", type=int)

client = tornadis.Client()
client.connect()

r = redis.StrictRedis()

class TestData(tornado.web.RequestHandler):
    @tornado.gen.coroutine
    def get(self):
        try:
            data = yield get_data()
            self.write(json_util.dumps(data))
            self.finish()
        except Exception as e:
            sys.stderr.write('Error -> %s' % e.message)

@tornado.gen.coroutine
def get_data():
    alerts = yield client.call("ZRANGE", "KEY", -5000, -1)
    raise tornado.gen.Return( alerts )

class Application(tornado.web.Application):
    def __init__(self):
        handlers = [
            (r'/', TestData),
        ]

        tornado_settings = dict(
            debug=True,
            auto_reload=True,
        )
        tornado.web.Application.__init__(self, handlers, **tornado_settings)

def main():
    tornado.options.parse_command_line()
    http_server = tornado.httpserver.HTTPServer( Application() )
    http_server.listen(options.port)
    tornado.ioloop.IOLoop.instance().start()

if __name__ == '__main__':
    main()

###


def add_test_data():
    pipe = r.pipeline(False)
    for score in xrange(5000):
        pipe.zadd("KEY", score, "testing_"+str(score))
    pipe.execute()
```

**Comment by thefab** [comment]
thanks for the complete bug report :-)

**Comment by thefab** [comment]
fixed in 0.2.2 release in pypi

**Comment by vgoklani** [comment]
Thank you so much !!! 

:)
